src/docfx/lib/git/GitRepoInfoProvider.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class GitRepoInfoProvider
+    {
+        private static readonly ConcurrentDictionary<string, GitRepoInfo> s_cache
```

Comment:
Don't use mutable static global state. In the extreme case, s_cache is wrong if submodules are updated when running `docfx watch`

Reply:
Suggest having a separate `BuildGitContributionInfo` step that runs before `BuildFiles`. That step returns a map of all the git contributors + edit url for each markdown/yaml file in build scope. Then `BuildMarkdown` can just lookup `gitContributionMap` just like looking up `tocMap`